dyno-jedis/src/main/java/com/netflix/dyno/jedis/operation/MultiKeyOperation.java

```diff
@@ -0,0 +1,57 @@
+package com.netflix.dyno.jedis.operation;
+
+import com.netflix.dyno.connectionpool.Operation;
+import com.netflix.dyno.jedis.OpName;
+import redis.clients.jedis.Jedis;
+
+import java.util.List;
+
+/**
+ * A poor man's solution for multikey operation. This is similar to
+ * basekeyoperation just that it takes a list of keys as arguments. For
+ * token aware, we just use the first key in the list. Ideally we should be
+ * doing a scatter gather
+ */
+public abstract class MultiKeyOperation<T> implements Operation<Jedis, T> {
+
+    private final List<String> keys;
+    private final List<byte[]> binaryKeys;
+    private final OpName op;
+
+    public MultiKeyOperation(final List keys, final OpName o) {
+        Object firstKey = (keys != null && keys.size() > 0) ? keys.get(0) : null;
+
+        if (firstKey != null) {
+            if (firstKey instanceof String) {//string key
+                this.keys = keys;
+                this.binaryKeys = null;
+            } else if (firstKey instanceof byte[]) {//binary key
+                this.keys = null;
+                this.binaryKeys = keys;
+            } else {//something went wrong here
+                this.keys = null;
+                this.binaryKeys = null;
+            }
+        } else {
+            this.keys = null;
+            this.binaryKeys = null;
+        }
+
+        this.op = o;
+    }
+
+    @Override
+    public String getName() {
+        return op.name();
+    }
+
+    @Override
+    public String getStringKey() {
```

Comment:
Would this be a source of confusion? Since we're sending back only the name of the first key? Maybe a comment about why this is the case would be useful.

Reply:
For some context, this was a refactor where I pulled these classes out from DynoJedisClient since they're useful outside also. Will add some comments.